hibernate-core/src/main/java/org/hibernate/persister/entity/EntityPersister.java

```diff
@@ -373,15 +373,23 @@ Serializable loadEntityIdByNaturalId(
 	/**
 	 * Load an instance of the persistent class.
 	 */
-	Object load(Serializable id, Object optionalObject, LockMode lockMode, SharedSessionContractImplementor session)
+	Object load(Serializable id, Object optionalObject, LockMode lockMode, SharedSessionContractImplementor session, Boolean readOnly)
 	throws HibernateException;
 
+	default Object load(Serializable id, Object optionalObject, LockMode lockMode, SharedSessionContractImplementor session) {
```

Comment:
The default implementation should actually be the one with the new signature in order to remain compatibility with custom persisters.

Reply:
Exactly! Thanks.